### PR TITLE
Google login

### PR DESCRIPTION
**Pull request by SaraVieira**
oh yeah, its happening

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/4642/commits/6689f8e41efa623bada105e82a4fc15a6dc75569 failed.

File: packages/app/src/app/pages/SignIn/components/Button.tsx [hunk 1]
```diff
@@ -3,10 +3,15 @@ import React from 'react';
 import { Button as BaseButton } from '@codesandbox/components';
 import { css } from '@styled-system/css';
 
-export const Button = ({ children, loading, ...props }) => (
+type Props = {
+  children: any;
+  loading?: boolean;
+  onClick: any;
```
Comment (CompuIves) [inline packages/app/src/app/pages/SignIn/components/Button.tsx:9]:
```suggestion
  onClick: () => void;
```

File: packages/app/src/app/pages/SignIn/components/Button.tsx [hunk 2]
```diff
@@ -3,10 +3,15 @@ import React from 'react';
 import { Button as BaseButton } from '@codesandbox/components';
 import { css } from '@styled-system/css';
 
-export const Button = ({ children, loading, ...props }) => (
+type Props = {
+  children: any;
```
Comment (CompuIves) [inline packages/app/src/app/pages/SignIn/components/Button.tsx:7]:
```suggestion
  children: React.ReactNode;
```

File: packages/app/src/app/overmind/actions.ts [hunk 3]
```diff
@@ -116,16 +116,22 @@ export const toggleSignInModal: Action = ({ state }) => {
 };
 
 export const signInButtonClicked: AsyncAction<{
-  useExtraScopes: boolean;
-} | void> = async ({ actions, state }, options) => {
-  if (!options) {
+  useExtraScopes?: boolean;
+  google?: boolean;
```
[the remark this answers is not in the document]
Comment (SaraVieira) [reply packages/app/src/app/overmind/actions.ts]: Done :)

File: packages/app/src/app/overmind/internalActions.ts [hunk 4]
```diff
@@ -142,6 +146,44 @@ export const signInGithub: AsyncAction<{ useExtraScopes?: boolean }, any> = (
...
+
+        document.cookie = `signedInDev=true; expires=${expiryDate.toUTCString()}; path=/`;
+      } else if (data?.jwt) {
+        effects.api.revokeToken(data.jwt);
+      }
+      popup.close();
+    });
+
+  effects.browser.waitForMessage('duplicate').then((data: any) => {
+    window.location.href = data.url + `&continue=/dashboard`;
```
[the remark this answers is not in the document]
Comment (SaraVieira) [reply packages/app/src/app/overmind/internalActions.ts]: Removed in refactor

**Pull request merged**